### PR TITLE
Features/action script linux transform

### PR DESCRIPTION
**Pull request by uzanh**

### CODE REVIEW

File: Ginger/GingerCoreNET/ActionsLib/ActScript.cs [hunk 1]
```diff
@@ -198,6 +200,11 @@ public override void Execute()
                         p.StartInfo.FileName = System.IO.Path.Combine(p.StartInfo.WorkingDirectory, ScriptName);
                         p.StartInfo.Arguments = Params;
                     }
+                    else if (ScriptInterpreterType == eScriptInterpreterType.SH)
+                    {
+                        string filePath = amdocs.ginger.GingerCoreNET.WorkSpace.Instance.SolutionRepository.ConvertSolutionRelativePath(Path.Combine(SolutionFolder, @"Documents/scripts/") + ScriptName);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/ActionsLib/ActScript.cs:205]: why to mention again "Documents/script" if you handled it in p.StartInfo.WorkingDirectory?

File: Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActScriptTests.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            Assert.AreEqual(actScript.ReturnValues.Count, 3);
+            Assert.AreEqual(string.Join(',', actScript.ActReturnValues.Select(x => x.Param).ToList()), "Var1,Var2,sum");
+            Assert.AreEqual(string.Join(',', actScript.ActReturnValues.Select(x => x.Actual).ToList()), "56,77,133");
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void BATFile1ArgTest()
+        {
+            //Arrange
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActScriptTests.cs:54]: I see you added seperated Unit tests so try adding Retrun on begining of it if the OS version is Windows and same for VBS if it is NOT Windows

**Review by MeniKadosh1 — CHANGES_REQUESTED** [review]

File: Ginger/Ginger/Actions/ActionEditPages/ActScriptEditPage.xaml.cs [hunk 3]
```diff
@@ -105,13 +108,13 @@ private void parseScriptHeader(string[] script)
         {
             foreach (string line in script)
             {
-                if (line.StartsWith("'GINGER_Description") || line.StartsWith("//GINGER_Description") || line.StartsWith("#GINGER_Description") || line.StartsWith("REM GINGER_Description"))
+                if (line.Contains("GINGER_Description"))
```
[the remark this answers is not in the document]
Comment (uzanh) [reply Ginger/Ginger/Actions/ActionEditPages/ActScriptEditPage.xaml.cs:111]: yes, it done in the regex with the sigh ^ (startwith), will recheck it

File: Ginger/GingerCoreNET/ActionsLib/ActScript.cs (hunk 1 above)
Comment (uzanh) [reply Ginger/GingerCoreNET/ActionsLib/ActScript.cs:205]: fixed

File: Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActScriptTests.cs (hunk 2 above)
Comment (uzanh) [reply Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActScriptTests.cs:54]: you want only one test sperate to win/linux tests?

**Comment by MeniKadosh1** [comment]
You keep them separated, just add a check for the OS so if it can’t be executed on it then the Unit test will be skipped (try adding “retrun” let’s see how it will be shown).

Regards,

Meni Kadosh
Software Development Team Lead
Amdocs Quality Engineering Services
R&D Center
+972-8-62-40961 (office)

       [signature_303716033]
Taking the industry to the cloud with the new cloud-native Amdocs CES20<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.amdocs.com%2Famdocsone%2Fces20&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152844403&sdata=JHW%2BIPUfWtT%2Fq%2BnfVxXWmbfBvGNIQRkfBFtKIAMpT%2BU%3D&reserved=0> suite and rich set of cloud practices
Read the latest on Amdocs.com<https://eur01.safelinks.protection.outlook.com/?url=Amdocs.com&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152854392&sdata=7rIOMY0C6C6SaPZBLm%2FykYv7IcCCgNQyqzl%2BrWRrGq4%3D&reserved=0> and the Amdocs blog network<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.amdocs.com%2Fblog&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152859376&sdata=9VD%2FhV8LDPdFkkwIVDq0afowNJFw4EL%2B94elyFldQWg%3D&reserved=0> – and follow us on Facebook<https://eur01.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.facebook.com%2FAmdocs&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152864367&sdata=kpMrwWkf7tLLD4y%2FTR85Y2XDs5aGmn0lp8ApkmPE0h0%3D&reserved=0>, Twitter<https://eur01.safelinks.protection.outlook.com/?url=http%3A%2F%2Ftwitter.com%2FAmdocs&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152869361&sdata=8TDEDSw5WHx3ynpuOt4WNac4JUhtEflNqsJW5O1pLyI%3D&reserved=0>, LinkedIn<https://eur01.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.linkedin.com%2Fcompany%2Famdocs&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152874349&sdata=H4N8taZRmbaRhJ7fHF0xNR0VmsBTorNxVmAIizCUPuc%3D&reserved=0> and YouTube<https://eur01.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.youtube.com%2Famdocs&data=02%7C01%7CMarios.Hajiloizi%40amdocs.com%7C461f79ffff6a41f5bff808d7ae3d6cc3%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637169451152879338&sdata=BUiWPmvCI%2BZteCd9s4Rdq0JwDt5KEeMYHrtLvAPWPhI%3D&reserved=0>.

From: Haim uzan ***@***.***>
Date: Thursday, 17 June 2021 at 13:57
To: Ginger-Automation/Ginger ***@***.***>
Cc: Meni Kadosh ***@***.***>, Review requested ***@***.***>
Subject: Re: [Ginger-Automation/Ginger] Features/action script linux transform (#1957)

@uzanh commented on this pull request.

________________________________

In Ginger/GingerCoreNETUnitTest/LinuxTransformationTests/ActScriptTests.cs<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGinger-Automation%2FGinger%2Fpull%2F1957%23discussion_r653455399&data=04%7C01%7CMeni.Kadosh%40amdocs.com%7Cf2487a1ee6694595af0308d9317ec026%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637595242744251070%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XTt1k5NgQdF5V%2B91SuR2%2B6cxyKQWBLZGoGrLNleXCso%3D&reserved=0>:

> +            actScript.AddNewReturnParams = true;

+

+            //Act

+            actScript.Execute();

+

+            //Assert;

+            Assert.AreEqual(actScript.ReturnValues.Count, 3);

+            Assert.AreEqual(string.Join(',', actScript.ActReturnValues.Select(x => x.Param).ToList()), "Var1,Var2,sum");

+            Assert.AreEqual(string.Join(',', actScript.ActReturnValues.Select(x => x.Actual).ToList()), "56,77,133");

+        }

+

+        [TestMethod]

+        [Timeout(60000)]

+        public void BATFile1ArgTest()

+        {

+            //Arrange

you want only one test sperate to win/linux tests?

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FGinger-Automation%2FGinger%2Fpull%2F1957%23discussion_r653455399&data=04%7C01%7CMeni.Kadosh%40amdocs.com%7Cf2487a1ee6694595af0308d9317ec026%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637595242744251070%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=XTt1k5NgQdF5V%2B91SuR2%2B6cxyKQWBLZGoGrLNleXCso%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAJTBJLR2DT7K6HUTOWG66P3TTHIIBANCNFSM463FSDNQ&data=04%7C01%7CMeni.Kadosh%40amdocs.com%7Cf2487a1ee6694595af0308d9317ec026%7Cc8eca3ca127646d59d9da0f2a028920f%7C0%7C0%7C637595242744261027%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Ru67XYBuL%2BpXUA7JCTxVPRGmhv5pXFhwmsSiK1OK2Z0%3D&reserved=0>.
This email and the information contained herein is proprietary and confidential and subject to the Amdocs Email Terms of Service, which you may review at https://www.amdocs.com/about/email-terms-of-service <https://www.amdocs.com/about/email-terms-of-service>

**Review by MeniKadosh1 — APPROVED** [review]

**Review by MeniKadosh1 — APPROVED** [review]

**Pull request merged**